packages/app/src/lib/import.ts

```diff
@@ -244,9 +251,113 @@ export async function asLastPass(data: string): Promise<VaultItem[]> {
  * Checks if a given string represents a LastPass CSV file
  */
 export function isLastPass(data: string): boolean {
-    return data.split("\n")[0] === "url,username,password,extra,name,grouping,fav";
+    try {
+        return data.split("\n")[0] === "url,username,password,extra,name,grouping,fav";
+    } catch (error) {
+        return false;
+    }
 }
 
-export function guessFormat(data: string): ImportFormat | null {
-    return isPBES2Container(data) ? PBES2 : isPadlockV1(data) ? PADLOCK_LEGACY : isLastPass(data) ? LASTPASS : CSV;
+async function parse1PuxItem(accountName: string, vaultName: string, item: OnePuxItem['item']): Promise<VaultItem | undefined> {
+    if (!item) {
+        return;
+    }
+
+    const rowData = parseToRowData(item, [accountName, vaultName]);
+
+    if (!rowData) {
+        return;
+    }
+
+    const itemName = rowData.name;
+    const tags = rowData.tags.split(',');
+
+    if (item.trashed) {
+        tags.push('trashed');
+    }
+
+    let fields: Field[] = [
+        new Field({ name: $l("Username"), value: rowData.username, type: FieldType.Username }),
+        new Field({ name: $l("Password"), value: rowData.password, type: FieldType.Password }),
+        new Field({ name: $l("URL"), value: rowData.url, type: FieldType.Url }),
+    ];
+
+    if (rowData.notes) {
+        fields.push(new Field({ name: $l("Notes"), value: rowData.notes, type: FieldType.Note }));
+    }
+
+    for (const extraField of rowData.extraFields) {
+        if (extraField.type === 'totp') {
+            // Extract just the secret
+            try {
+                const secret = new URL(extraField.value).searchParams.get('secret');
+                if (secret) {
+                    fields.push(new Field({ name: extraField.name, value: secret, type: FieldType.Totp }));
+                }
+            } catch (error) {
+                // Do nothing
+            }
+        } else {
+            // @ts-ignore All of extraField.type possibilities match FieldType.*
+            fields.push(new Field({ name: extraField.name, value: extraField.value, type: extraField.type }));
+        }
+
+    }
+
+    return createVaultItem(itemName, fields, tags);
+}
+
+export async function as1Pux(file: string | Uint8Array): Promise<VaultItem[]> {
+    try {
+        const dataExport = await parse1PuxFile(file);
+
+        const items = [];
+        
+        for (const account of dataExport.data.accounts) {
+            for (const vault of account.vaults) {
```

Comment:
Idea: If there are multiple vaults, add the vault name as a tag? That way the items can easily be split up into multiple vaults later.

Reply:
I.e.: (further down)
```ts
if  (account.vaults.length) {
    parsedItem.tags.push(vault.attrs.name);
}
```

Reply:
In this line, the vault name is being passed along to be added as a tag already: https://github.com/padloc/padloc/pull/328/files/feb7a3ce532bf85ce2a4dfb5c8e25fbad5c0f4b2#diff-f83e7b908e8e3f2938f2ff2d2b0e7d15930ddcc1065909bdb3a0de0251aa7e01R320 

Let me know if something isn't clear, or if you meant something else.